app/controllers/csvs_controller.rb

```diff
@@ -0,0 +1,107 @@
+class CsvsController < ApplicationController
+
+  rescue_from ActiveRecord::RecordNotFound do |exception|
+    respond_to do |format|
+      format.json { render json: {status: "not found"}.to_json, status: :not_found }
+      format.csv  { render body: "not found", status: :not_found }
+      format.html { render file: "#{Rails.public_path}/404.html", status: :not_found }
+    end
+  end
+
+  def index
+    @csv_exports = CsvExport.where(user_id: current_user.id)
+    respond_to do |format|
+      format.html
+      format.json { render json: @csv_exports }
+    end
+  end
+  
+  def show
+    @csv_export = CsvExport.find(params[:id])
+    respond_to do |format|
+      format.html
+      format.json { render json: @csv_export }
+      format.csv { download }
+    end
+  end
+
+  def create
+    filters = deploy_filter(params)
+    csv_export = CsvExport.create!(user: current_user, filters: filters.to_json)
+    csv_export.status! :pending
```

Comment:
the csv_export can take care of this with a before_create callback / by having pending as a default value